packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/portal/services/PortalDescriptionProvider.java

```diff
@@ -54,12 +60,20 @@ public List<IRepresentationDescription> getRepresentationDescriptions(IEditingCo
                 .map(this.objectService::getId)
                 .orElse("");
 
+        Function<VariableManager, List<String>> iconURLsProvider = variableManager -> {
+            return this.representationImageProviders.stream()
+                        .map(representationImageProvider -> representationImageProvider.getImageURL(Portal.KIND))
+                        .flatMap(Optional::stream)
+                        .toList();
```

Comment:
It makes me wonder, what's the responsibility of the `IRepresentationImageProvider`? To be the source of truth or to be the fallback? Because on hand since it is used everywhere, it creates the expectation that everybody should do just that in all representation description and in this case the exception becomes that all persisted representations should have icon URLs in the database.

Yet, the lack of a migration for the database to fill those icons for existing representations and the use of this in the ExplorerDescriptionProviders seems to suggest that on the other hand those `IRepresentationImageProvider` are a fallback to be used when we have nothing in the database / description. On top of that, most of the existing descriptions (details, explorer, related elements, etc) don't bother creating proper `iconURLsProvider`.

Since most (all?) of the code necessary to consider `IRepresentationImageProvider` as a fallback in case of missing icon URLs is there, I don't see why any representation description should be bothered with such dependency. Either we should do it or specifiers should do it, it's cumbersome to do both.